src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -159,14 +159,17 @@ private boolean processCursor(final String subscriptionId, final SubscriptionCur
                 cursorToProcess.getPartition(),
                 cursorToProcess.getOffset());
 
-        topicRepository.validateCommitCursor(toProcess);
-        return commitCursor(subscriptionId, eventType.getTopic(), cursorToProcess);
+        timelineService.getTopicRepository(eventType).validateCommitCursor(toProcess);
+        return commitCursor(subscriptionId, eventType, cursorToProcess);
     }
 
-    private boolean commitCursor(final String subscriptionId, final String eventType, final SubscriptionCursor cursor)
+    private boolean commitCursor(final String subscriptionId,
+                                 final EventType eventType,
+                                 final SubscriptionCursor cursor)
             throws ServiceUnavailableException, NoSuchSubscriptionException, InvalidCursorException {
 
-        final String offsetPath = format(PATH_ZK_OFFSET, subscriptionId, eventType, cursor.getPartition());
+        final String topic = eventType.getTopic();
```

Comment:
I fixed it here aae9086, but again this is not fully correct approach to do it. The timeline has to be retrieved according to offset. I checked that EventType.getTopic is used in a lot of places, for example EventTypeController.delete, there should be removed topics for all timelines. In my opinion it has to addressed in the following PRs.